src/ui/layout/mct-tree.vue

```diff
@@ -330,105 +355,64 @@ export default {
                 this.updatingView = false;
             });
         },
-        async setContainerHeight() {
-            await this.$nextTick();
+        setContainerHeight() {
             let mainTree = this.$refs.mainTree;
             let mainTreeHeight = mainTree && mainTree.clientHeight ? mainTree.clientHeight : 0;
 
             if (mainTreeHeight !== 0) {
-                this.calculateChildHeight(() => {
-                    let ancestorsHeight = this.calculateAncestorHeight();
-                    let allChildrenHeight = this.calculateChildrenHeight();
-
-                    if (this.activeSearch) {
-                        ancestorsHeight = 0;
-                    }
-
-                    this.availableContainerHeight = mainTreeHeight - ancestorsHeight;
-
-                    if (allChildrenHeight > this.availableContainerHeight) {
-                        this.setPageThreshold();
-                        this.noScroll = false;
-                    } else {
-                        this.noScroll = true;
-                    }
-
-                    this.updatevisibleItems();
-                });
+                this.availableContainerHeight = mainTreeHeight - this.ancestorsHeight;
             } else {
                 window.setTimeout(this.setContainerHeight, RECHECK_DELAY);
             }
         },
+        childrenHeight() {
```

Comment:
Why is this a method rather than a computed property? Methods are really expensive when they're used from templates because they have to be re-evaluated every time.

Reply:
I think I have a fix for this, but need time to implement.

Reply:
So the issue here, was in children height I'm using the top margin of the tree. Since this doesn't really change, I just need to grab the final version on load, this was unreliable in mounted, apparently CSS was still updating after that hook, so I used some vanilla JS for readyState complete, that did the trick. Now childrenHeight is only calculated when item height or focused items changes.